llvm/lib/Target/AIE/AIE2AddrSpace.h

```diff
@@ -0,0 +1,46 @@
+//===-- AIE2AddrSpace.h - Define Address Space for AIEngine V2 ---*- C++-*-===//
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2023-2024 Advanced Micro Devices, Inc. or its affiliates
+//
+//===----------------------------------------------------------------------===//
+//
+// This file declares the AIEngine V2 Address Space and DM banks
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_SUPPORT_AIE2ADDRSPACE_H
+#define LLVM_SUPPORT_AIE2ADDRSPACE_H
+
+namespace llvm {
+
+namespace AIE2 {
+
+enum class AddressSpaces {
+  none, // Default address space
+  PM,   // Address space for Program Memory (PM)
+  DM,   // Address space for Data Memory(DM) includes Bank A, B, C, D
+  DM_test,
+  stack, // Address space for stack
```

Comment:
Are `DM_test` and `stack` actually used somewhere?

Reply:
"stack" could be used as per AIE-API, Not sure about DM_test but we need to have the same `enum` number for flows to work in future. We can discuss this in more detail.